card/lib/card/codename.rb

```diff
@@ -3,17 +3,22 @@
 require_dependency "card/name"
 
 class Card
-  # {Card}'s names can be changed, and therefore names should not be directly
-  # mentioned in code, lest a name change break the application. Instead, a
-  # {Card} that needs specific code manipulations should be given a {Codename},
-  # a unique string identifier that will not change even if the card's name
+  # {Card}'s names can be changed, and therefore _names_ should not be directly mentioned in code, lest a name change break the application.
+  #
+  # Instead, a {Card} that needs specific code manipulations should be given a {Codename}, which will not change even if the card's name
   # does.
   #
-  # The {Codename} class provides a fast cache for this slow-changing data.
-  # Every process maintains a complete cache that is not frequently reset
+  # An administrator might add to the Company card via the RESTful web API with a url like
+  #
+  #     /update/CARDNAME?card[codename]=CODENAME
+  #
+  # ...or via the api like
+  #
+  #     Card[CARDNAME].update_attributes! codename: CODENAME
+  #
+  # Generally speaking, _codenames_ are represented by Symbols.
```

Comment:
Since a decker will never use codenames, it's probably more important to mention how to add codenames with migrations.

Reply:
good point.  I've been more focused on making sure people can successfully get started on their own mod rather than best practices for deployment.

This is related to "Make mod installation handle card creation/updates" (Mod Squad doc).  If we did that well, then perhaps the best practice could also be quick/simple enough for a newbie.